bionic/flow.py

```diff
@@ -1986,13 +1978,24 @@ def core__aip_executor(
                 core__persistent_cache__gcs__fs is None.
             """
             raise AssertionError(oneline(error_message))
+
+        if core__aip_execution__docker_image_uri is None:
+            docker_image_uri_func = build_image_if_missing_async(
```

Comment:
Is it possible for `build_image_if_missing_async` to hang, or just take a long time? If so, it might be confusing to the user, since it won't be obvious from the log messages that Bionic is waiting for this thread to finish. WDYT about wrapping this function like this?

    docker_image_uri_future = build_image_if_missing_async( core__aip_execution__config.project_name)
    def docker_image_uri_func():
        if not docker_image_uri_future.done():
            logger.info("Waiting for Docker image to finish building...")
        return docker_image_uri_future.result